backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -796,11 +871,22 @@ public Project getProjectByID(String id) throws InvalidProtocolBufferException {
 
   @Override
   public Project setProjectShortName(String projectId, String projectShortName, UserInfo userInfo)
-      throws InvalidProtocolBufferException {
+      throws InvalidProtocolBufferException, ModelDBException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      Query query = session.createQuery(GET_PROJECT_BY_SHORT_NAME_AND_OWNER_HQL);
+      Set<String> accessibleProjectIds =
+          ModelDBUtils.filterWorkspaceOnlyAccessibleIds(
+              roleService,
+              new HashSet<>(Collections.singletonList(projectId)),
+              null,
+              userInfo,
+              ModelDBServiceResourceTypes.PROJECT);
+      if (accessibleProjectIds.isEmpty()) {
+        throw new ModelDBException("You can not set project short_name in others workspace");
```

Comment:
So here i am setting given project id `new HashSet<>(Collections.singletonList(projectId)),` so i am calling to the UAC with the worksapceName and the projectID. So if that projectId is not from my workspace then it should not be editable to me so UAC return empty resource list.

Reply:
Yes

Reply:
But `filterWorkspaceOnlyAccessibleIds` is not being passed any info about workspace, right? It's also possible that I can edit project in other workspaces (e.g. collaborator or in organization). Shouldn't we leave decisions of whether the user can access a resource or not to UAC? Spreading the logic around seems error prone, so I'm not sure we have a good reason to do it. Here it could just be a call to "isSelfAllowed" and it would be done (and it would be clearer what is the permission check). Maybe I'm missing something?

Reply:
Ok, got it your point now. `filterWorkspaceOnlyAccessibleIds` is not work here. i will revert the logic based on the `isSelfAllowed`

Reply:
Cool. We've built a lot of machinery in MDB that might not work the same and might be simplified by a quick call to UAC =) We'll catch them as we find them during code changes.

Reply:
updated code with isSelfAllowed endpoint